### PR TITLE
Select agents in agent group whe running new scan

### PR DESCRIPTION
**Pull request by adnaneserrar**
[Screencast from 12 غشت, 2024 +01 11:43:34.webm](https://github.com/user-attachments/assets/216fd10a-77d0-424f-9f04-68ee74cfdcc0)

### CODE REVIEW

File: project/agent-groups/components/AgentGroupForm.vue [hunk 1]
```diff
@@ -232,8 +233,12 @@ const getScannerDetails = (scanner: Scanner): { title: string, subtitle: string
 // Watch for changes in the prop agentGroup and update the localAgentGroup.
 watch(() => props.agentGroup, (newVal) => {
   localAgentGroup.value = { ...newVal } as OxoAgentGroupType
+  yamlSource.value = props.agentGroup?.yamlSource || ''
 }, { immediate: true, deep: true })
 
+watch(() => yamlSource, (newVal) => {
+  yamlSource.value = newVal.value
+}, { immediate: true, deep: true })
```
Comment (oussamaessaji) [inline project/agent-groups/components/AgentGroupForm.vue:241]:
``` suggestion
watch(() => yamlSource, (newVal) => {
  yamlSource.value = newVal.value
}, { immediate: true, deep: true })
```

**Review by nmasdoufi-ol — APPROVED** [review]
LGTM from the video.

File: project/scans/components/AgentGroupSelect.vue [hunk 2]
```diff
@@ -203,7 +200,14 @@ export default defineComponent({
      */
       try {
         await this.getAgentGroup()
+        const agentGroup = await this.updateAgentGroup()
+        console.log(agentGroup)
```
Comment (oussamaessaji) [inline project/scans/components/AgentGroupSelect.vue:204]:
```suggestion
```

File: project/scans/components/AgentGroupSelect.vue [hunk 3]
```diff
@@ -216,6 +219,58 @@ export default defineComponent({
       } catch (e: any) {
         this.reportError(e?.message || 'Error getting agent group')
       }
+    },
+    getAgentGroupInput(yamlSource: string | null) {
```
Comment (3asm) [inline project/scans/components/AgentGroupSelect.vue:223]: Docstring.

File: project/scans/components/AgentGroupSelect.vue [hunk 4]
```diff
@@ -216,6 +219,58 @@ export default defineComponent({
...
+      if (yamlSource === null || yamlSource === undefined) {
+        return
+      } else {
+        const agentGroupInput = Yaml.stringify(Yaml.parse(yamlSource))
+        if (agentGroupInput !== null) {
+          return agentGroupInput
+        }
+      }
+    },
+    async updateAgentGroup(): Promise<OxoAgentGroupType | null | undefined> {
```
Comment (3asm) [inline project/scans/components/AgentGroupSelect.vue:233]: Docstrin.

File: project/scans/components/AgentGroupSelect.vue [hunk 5]
```diff
@@ -216,6 +219,58 @@ export default defineComponent({
...
+          return this.agentGroupService.createAgentGroup({
+            scanner: this.selectedScanner,
+            agentGroup: {
+              description: agentGroupDefinition.description || description,
+              agents: agentGroupDefinition?.agents,
+              name: agentGroupDefinition.name || name,
+              assetTypes: [this.agentGroupAssetType]
+            }
+          })
+        }
```
Comment (3asm) [inline project/scans/components/AgentGroupSelect.vue:254]: Inlined functions are not a good practice, why not get it out?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: project/scans/components/AgentGroupSelect.vue [hunk 6]
```diff
@@ -216,6 +219,69 @@ export default defineComponent({
...
+ * Updates the agent group by either creating a new one or updating an existing one.
+ * */
+    async updateAgentGroup(): Promise<OxoAgentGroupType | null | undefined> {
+      if (!this.agentGroupInput) return
+
+      try {
+        const agentGroupInput = this.getAgentGroupInput(this.agentGroupInput)
+        const agentGroupDefinition = Yaml.parse(agentGroupInput)
+        if (!this.selectedScanner || !agentGroupInput) return
+        if (this.selectedAgentGroup && this.selectedAgentGroup.length > 0) {
```
Comment (elyousfi5) [inline project/scans/components/AgentGroupSelect.vue:248]: please use explicit checks

File: project/scans/components/AgentGroupSelect.vue [hunk 7]
```diff
@@ -203,19 +200,88 @@ export default defineComponent({
...
+        if (agentGroupInput !== null) {
+          return agentGroupInput
+        }
+      }
+    },
+    /**
+ * Updates the agent group by either creating a new one or updating an existing one.
+ * */
+    async updateAgentGroup(): Promise<OxoAgentGroupType | null | undefined> {
+      if (!this.agentGroupInput) return
```
Comment (ybadaoui-ostorlab) [inline project/scans/components/AgentGroupSelect.vue:242]: explicit check here!

**Review by ybadaoui-ostorlab — APPROVED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**